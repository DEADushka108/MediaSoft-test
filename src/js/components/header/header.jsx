import React from 'react';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';
import { setCartList } from '../../store/ships/selectors';
import { AppRoute } from '../../utils/const';
import PropTypes from 'prop-types';

const Header = (props) => {
  const {shipsList} = props;

  return (
    <header className="page-header">
      <div className="page-header__wrapper">
        <Link to={AppRoute.ROOT} className="page-header__logo">
          Starshipsshop
        </Link>
        <nav className="main-nav">
          <ul className="main-nav__list">
            <li className="main-nav__item">
              <Link to={AppRoute.ROOT} className="main-nav__link main-nav__link--active">
                <svg className="main-nav__icon">
                  <use xlinkHref="#home-page"></use>
                </svg>
                <span className="main-nav__text">Shop</span>
              </Link>
            </li>
            <li className="main-nav__item">
              <Link to={AppRoute.CONTACT} className="main-nav__link">
              <svg className="main-nav__icon">
                  <use xlinkHref="#social-group"></use>
              </svg>
              <span className="main-nav__text">Contact</span>
              </Link>
            </li>
            <li className="main-nav__item">
              <Link to={AppRoute.CART} className="main-nav__link">
              <svg className="main-nav__icon">
                  <use xlinkHref="#shopping-cart"></use>
                </svg>
                <span className="main-nav__text">Cart</span>
                <span className="main-nav__sub">{shipsList.length}</span>
              </Link>
            </li>
          </ul>
        </nav>
      </div>
    </header>
  );
}

Header.propTypes = {};

const mapStateToProps = (state) => ({
  shipsList: setCartList(state),
})

export {Header}
export default connect(mapStateToProps)(Header);