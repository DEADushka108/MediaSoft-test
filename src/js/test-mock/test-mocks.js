export * from './const/const';
export * from './ships/ships';
export * from './store/store';
