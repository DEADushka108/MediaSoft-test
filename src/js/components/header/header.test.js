import React from 'react';
import renderer from 'react-test-renderer';
import {MemoryRouter} from 'react-router-dom';
import {Provider} from 'react-redux';
import {store, shipsList} from '../../test-mock/test-mocks';
import {Header} from './header';

it(`Should Header component render correctly`, () => {
  const tree = renderer
    .create((
      <Provider store={store}>
        <MemoryRouter>
          <Header
            shipsList={shipsList}
          />
        </MemoryRouter>
      </Provider>
    ), {
      createNodeMock() {
        return {};
      }
    })
    .toJSON();

  expect(tree).toMatchSnapshot();
});
