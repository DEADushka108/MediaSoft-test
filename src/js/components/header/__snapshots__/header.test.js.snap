// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should Header component render correctly 1`] = `
<header
  className="page-header"
>
  <div
    className="page-header__wrapper"
  >
    <a
      className="page-header__logo"
      href="/"
      onClick={[Function]}
    >
      Starshipsshop
    </a>
    <nav
      className="main-nav"
    >
      <ul
        className="main-nav__list"
      >
        <li
          className="main-nav__item"
        >
          <a
            className="main-nav__link main-nav__link--active"
            href="/"
            onClick={[Function]}
          >
            <svg
              className="main-nav__icon"
            >
              <use
                xlinkHref="#home-page"
              />
            </svg>
            <span
              className="main-nav__text"
            >
              Shop
            </span>
          </a>
        </li>
        <li
          className="main-nav__item"
        >
          <a
            className="main-nav__link"
            href="/contact"
            onClick={[Function]}
          >
            <svg
              className="main-nav__icon"
            >
              <use
                xlinkHref="#social-group"
              />
            </svg>
            <span
              className="main-nav__text"
            >
              Contact
            </span>
          </a>
        </li>
        <li
          className="main-nav__item"
        >
          <a
            className="main-nav__link"
            href="/cart"
            onClick={[Function]}
          >
            <svg
              className="main-nav__icon"
            >
              <use
                xlinkHref="#shopping-cart"
              />
            </svg>
            <span
              className="main-nav__text"
            >
              Cart
            </span>
            <span
              className="main-nav__sub"
            >
              7
            </span>
          </a>
        </li>
      </ul>
    </nav>
  </div>
</header>
`;
